website/src/components/canvas/CanvasView.jsx

```diff
@@ -49,17 +44,19 @@ class CanvasView extends React.Component {
    */
   componentDidMount() {
     firebase.database().ref('/test').once('value').then((elemListSnap) => {
-      this.props.dispatch(initElements(elemListSnap.val()));
+      this.props.dispatch(ElementActions.initElements(elemListSnap.val()));
+    });
+    firebase.database().ref('/test').on('child_added', (elemSnap) => {
+      this.props.dispatch(ElementActions.addElement(elemSnap.key,
+        elemSnap.val()));
     });
-
     firebase.database().ref('/test').on('child_changed', (elemSnap) => {
-      this.props.dispatch(updateElement(UPDATE_POSITION, elemSnap.key,
-        elemSnap.child('position').val()));
-      this.props.dispatch(updateElement(UPDATE_SIZE, elemSnap.key,
-        elemSnap.child('size').val()));
-      this.props.dispatch(updateElement(UPDATE_ROTATION, elemSnap.key,
-        Number(elemSnap.child('rotation').val())));
+      this.props.dispatch(ElementActions.addElement(elemSnap.key,
```

Comment:
It is primarily for adding though since it will set a new one in the list of elements. I use it here just because add overwrites what was there before.